### PR TITLE
Suggestion: registry as set

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Say severe times in the segmentation key several times so would suggest using set instead of the list...
```
 'efficientnet-b0',
 'efficientnet-b0',
 'efficientnet-b1',
 'efficientnet-b1',
 'efficientnet-b2',
 'efficientnet-b2',
 'efficientnet-b3',
 'efficientnet-b3',
 'efficientnet-b4',
 'efficientnet-b4',
 'efficientnet-b5',
 'efficientnet-b5',
 'efficientnet-b6',
 'efficientnet-b6',
 'efficientnet-b7',
 'efficientnet-b7',
 'efficientnet-b8',
 'efficientnet-l2',
```

## Before submitting
- [ ] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by karthikrangasai** [comment]
Hello @Borda ,

I am working on #1346 which freezes the versions of the `protobuf` and `omegaconf` to avoid the current failing tests.
Once the tests pass there, and the PR is merged, this one will clear up.

File: flash/core/registry.py [hunk 1]
```diff
@@ -309,4 +309,4 @@ def _register_function(
                 return registry._register_function(fn, name=name, override=override, metadata=metadata)
 
     def available_keys(self) -> List[str]:
-        return list(itertools.chain.from_iterable(registry.available_keys() for registry in self.registries))
+        return list(itertools.chain.from_iterable(registry.available_keys() for registry in {self.registries}))
```
Comment (krshrimali) [inline flash/core/registry.py:312]:
```suggestion
        return list(itertools.chain.from_iterable(registry.available_keys() for registry in set(self.registries)))
```

File: flash/image/segmentation/model.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any, Dict, List, Optional, Type, Union
+from typing import Any, Dict, List, Optional, Set, Type, Union
```
Comment (krshrimali) [inline flash/image/segmentation/model.py:14]: pep8: Unused import

```suggestion
from typing import Any, Dict, List, Optional, Type, Union
```

File: flash/image/embedding/model.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import warnings
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Set, Optional
```
Comment (krshrimali) [inline flash/image/embedding/model.py]: pep8: Unused import

```suggestion
from typing import Any, Dict, Set, Optional
```

File: flash/core/registry.py [hunk 4]
```diff
@@ -14,7 +14,7 @@
 import functools
 import inspect
 import itertools
-from typing import Any, Callable, Dict, List, Optional, Union
+from typing import Any, Callable, Dict, List, Optional, Set, Union
```
Comment (krshrimali) [inline flash/core/registry.py:17]: pep8: Unused import

```suggestion
from typing import Any, Callable, Dict, List, Optional, Union
```

**Review by krshrimali — APPROVED** [review]
Thanks, @Borda for the PR. And sorry to keep you waiting for this PR (and a few others).

Overall, looks good to me. Left a few suggestions to fix pep8 failures, and I hope it's fine if I commit them myself, considering you would be occupied. :) If it's not, please feel free to let me know and we can revert.

Thanks!

File: flash/image/embedding/model.py [hunk 5]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import warnings
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Optional, Set
```
Comment (krshrimali) [inline flash/image/embedding/model.py:15]:
```suggestion
from typing import Any, Dict, Optional, Set
```

**Comment by krshrimali** [comment]
Hi, @Borda - Thanks for this PR, but after looking at the failing tests, and thinking about it further, I think it's better to follow a general behavior - if the user adds something twice, it will appear twice. They can always use a set to get unique out of it, but they can't do the reverse. I'm closing this PR in favor of this thought, however - I'm happy to discuss it further if you have any thoughts or think differently. Thanks!

**Comment by Borda** [comment]
> if the user adds something twice, it will appear twice

That is true, but how a user can select `efficientnet-b0`, not the first one?
I agree that if a user adds something, it is his responsibility, but this is default Flash, our responsibility... :)
I think that @ethanwharris had a theory that there were coming  from different sources like one trained with TF init and the other with PT init :rabbit: 
https://discuss.pytorch.org/t/suboptimal-convergence-when-compared-with-tensorflow-model/5099/39

**Pull request closed without merging**